x-pack/legacy/plugins/reporting/export_types/png/server/execute_job/index.ts

```diff
@@ -29,22 +22,21 @@ type QueuedPngExecutorFactory = ExecuteJobFactory<ESQueueWorkerExecuteFn<JobDocP
 
 export const executeJobFactory: QueuedPngExecutorFactory = async function executeJobFactoryFn(
   reporting: ReportingCore,
-  server: ServerFacade,
-  elasticsearch: ElasticsearchServiceSetup,
   parentLogger: Logger
 ) {
+  const config = await reporting.getConfig();
   const browserDriverFactory = await reporting.getBrowserDriverFactory();
-  const generatePngObservable = generatePngObservableFactory(server, browserDriverFactory);
+  const generatePngObservable = generatePngObservableFactory(config, browserDriverFactory);
   const logger = parentLogger.clone([PNG_JOB_TYPE, 'execute']);
 
   return function executeJob(jobId: string, job: JobDocPayloadPNG, cancellationToken: any) {
     const jobLogger = logger.clone([jobId]);
     const process$: Rx.Observable<JobDocOutput> = Rx.of(1).pipe(
-      mergeMap(() => decryptJobHeaders({ server, job, logger })),
+      mergeMap(() => decryptJobHeaders({ config, job, logger })),
```

Comment:
Similar to my comment above: maybe we just pass the properties `decryptJobHeaders` cares about, instead of having that helper deal with the config interface?